File: untitled-user/task.py
import os

import argparse
arg_parser = argparse.ArgumentParser()

arg_parser.add_argument('--id', action='store', type=str, required=True, dest='id')


arg_parser.add_argument('--names', action='store', type=str, required=True, dest='names')

arg_parser.add_argument('--param_p', action='store', type=str, required=True, dest='param_p')

args = arg_parser.parse_args()
print(args)

id = args.id

import json
names = json.loads(args.names)

param_p = args.param_p


for name in names:
    print(f"Hello, {name}!")
x = 1
print(param_p)
print(os.getenv('API_ENDPOINT'))

import json
filename = "/tmp/x_" + id + ".json"
file_x = open(filename, "w")
file_x.write(json.dumps(x))
file_x.close()

File: process-name-list-abr-rstudio-local/task.R
setwd('/app')

# retrieve input parameters

library(optparse)
library(jsonlite)
if (!requireNamespace("dplyr", quietly = TRUE)) {
	install.packages("dplyr", repos="http://cran.us.r-project.org")
}
library(dplyr)
if (!requireNamespace("stats", quietly = TRUE)) {
	install.packages("stats", repos="http://cran.us.r-project.org")
}
library(stats)


option_list = list(

make_option(c("--id"), action="store", default=NA, type="character", help="my description"), 
make_option(c("--names"), action="store", default=NA, type="character", help="my description"), 
make_option(c("--param_p"), action="store", default=NA, type="character", help="my description"), 
make_option(c("--x"), action="store", default=NA, type="integer", help="my description"), 
make_option(c("--y"), action="store", default=NA, type="integer", help="my description")

)

# set input parameters accordingly
opt = parse_args(OptionParser(option_list=option_list))

id <- gsub('"', '', opt$id)
names = fromJSON(opt$names)
x = opt$x
y = opt$y

param_p = opt$param_p




for (name in names) {
  print(paste0('Hello, ', name))
}
t <- x + y
m <- stats::median(c(1, 2))
print(Sys.getenv())
print(param_p)



# capturing outputs
file <- file(paste0('/tmp/t_', id, '.json'))
writeLines(toJSON(t, auto_unbox=TRUE), file)
close(file)
